lib/vagrant-libvirt/util/network_util.rb

```diff
@@ -48,7 +49,7 @@ def configured_networks(env, logger)
                   error_message: "#{management_network_address} does not include both an address and subnet mask"
           end
 
-          if qemu_use_session
+          if qemu_use_session or qemu_use_agent == true
```

Comment:
I've a query on whether this is necessary or whether we should be able to use the qemu guest agent to retrieve the IP address for any guess that has it installed and enabled, and make the decision on what network is used for the management network to be handled explicitly rather than bundled as part of this option. Probably better to expand on this via the issue as there is more context there.